### PR TITLE
Dig into language choice of the base project template

### PR DESCRIPTION
**Pull request by amorde**
Closes https://github.com/CocoaPods/CocoaPods/issues/8063

I'm not sure what the template was in Xcode 9, but Xcode 10's base project template now has build settings specific to language choice, but only for Swift.

Here's the `Options` section of the base project template in Xcode 10

```xml
<key>Options</key>
    <array>
        <dict>
            <key>Identifier</key>
            <string>languageChoice</string>
            <key>Units</key>
            <dict>
                <key>Swift</key>
                <dict>
                    <key>Project</key>
                    <dict>
                        <key>Configurations</key>
                        <dict>
                            <key>Debug</key>
                            <dict>
                                <key>SWIFT_OPTIMIZATION_LEVEL</key>
                                <string>-Onone</string>
                                <key>SWIFT_ACTIVE_COMPILATION_CONDITIONS</key>
                                <string>DEBUG</string>
                            </dict>
                            <key>Release</key>
                            <dict>
                                <key>SWIFT_COMPILATION_MODE</key>
                                <string>wholemodule</string>
                                <key>SWIFT_OPTIMIZATION_LEVEL</key>
                                <string>-O</string>
                            </dict>
                        </dict>
                    </dict>
                    <key>Targets</key>
                    <array>
                        <dict>
                            <!-- Declare basic stuff that is true of all Swift targets, but mark this non-concrete so it won't cause actual creation of a target. -->
                            <key>Concrete</key>
                            <false/>
                            <key>SharedSettings</key>
                            <dict>
                                <key>SWIFT_VERSION</key>
                                <string>4.2</string>
                            </dict>
                        </dict>
                    </array>
                </dict>
            </dict>
        </dict>
    </array>
```

because we weren't digging into the `Swift` language choice, we weren't including these settings in the base project config.

I think eventually we'll need to split up these templates into 2 versions or something, but for now this fixes the issue.

### CODE REVIEW

**Comment by amorde** [comment]
Looks like there's some tests that need to be updated. Will look into it

**Comment by amorde** [comment]
oh, right. should probably actually update the constants rather than just the specs 🤦‍♂️

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Dig into language choice of the base project template  
  [amorde](https://github.com/amorde)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: spec/fixtures/CommonBuildSettings/rebuild_from_xcode [hunk 1]
```diff
@@ -106,7 +106,7 @@ end.compact
 puts "\r\033[KRead all templates"
 
 targets = {
-  'Project'                        => { 'languageChoice' => nil,            :template_identifier => 'com.apple.dt.unit.base' },
```
Comment (segiddins) [inline spec/fixtures/CommonBuildSettings/rebuild_from_xcode]: 🤦‍♂️  Thank you

**Pull request merged**